src/pages/_app.tsx

```diff
@@ -0,0 +1,130 @@
+import useEagerConnect from 'hooks/useEagerConnect'
+import { useInactiveListener } from 'hooks/useInactiveListener'
+import useSentryUser from 'hooks/useSentryUser'
+import useUserAgent from 'hooks/useUserAgent'
+import type { AppProps } from 'next/app'
+import Head from 'next/head'
+import React, { Fragment } from 'react'
+import { PersistGate } from 'redux-persist/integration/react'
+import { persistor } from 'state'
+import { usePollBlockNumber } from 'state/block/hooks'
+import { usePollCoreFarmData } from 'state/farms/hooks'
+import { NextPage } from 'next'
+import { useFetchProfile } from 'state/profile/hooks'
+import { Blocklist, Updaters } from '..'
+import Menu from '../components/Menu'
+import Providers from '../Providers'
+import GlobalStyle from '../style/Global'
+
+// This config is required for number formatting
+BigNumber.config({
+  EXPONENTIAL_AT: 1000,
+  DECIMAL_PLACES: 80,
+})
+
+function GlobalHooks() {
+  usePollBlockNumber()
+  useEagerConnect()
+  useFetchProfile()
+  usePollCoreFarmData()
+  useUserAgent()
+  useInactiveListener()
+  useSentryUser()
+  return null
+}
+
+function MyApp(props: AppProps) {
+  return (
+    <>
+      <Head>
+        <meta
+          name="viewport"
+          content="width=device-width, initial-scale=1, maximum-scale=5, minimum-scale=1, viewport-fit=cover"
+        />
+        <meta
+          name="description"
+          content="Cheaper and faster than Uniswap? Discover PancakeSwap, the leading DEX on Binance Smart Chain (BSC) with the best farms in DeFi and a lottery for CAKE."
+        />
+        <meta name="theme-color" content="#1FC7D4" />
+        <meta name="twitter:image" content="https://pancakeswap.finance/images/hero.png" />
+        <meta
+          name="twitter:description"
+          content="The most popular AMM on BSC! Earn CAKE through yield farming or win it in the Lottery, then stake it in Syrup Pools to earn more tokens! Initial Farm Offerings (new token launch model pioneered by PancakeSwap), NFTs, and more, on a platform you can trust."
+        />
+        <meta name="twitter:card" content="summary_large_image" />
+        <meta
+          name="twitter:title"
+          content="🥞 PancakeSwap - A next evolution DeFi exchange on Binance Smart Chain (BSC)"
+        />
+        <title>PancakeSwap</title>
+      </Head>
+      <Providers>
+        <Blocklist>
+          <GlobalHooks />
+          <Updaters />
+          <ResetCSS />
+          <GlobalStyle />
+          <GlobalCheckClaimStatus excludeLocations={[]} />
+          {/*
+            Conditionally rendering `PersistGate` will cause react dom hydrate error.
+            "Warning: Did not expect server HTML to contain a <div> in <div>."
+           */}
+          {process.browser ? (
+            <PersistGate loading={null} persistor={persistor}>
+              <App {...props} />
+            </PersistGate>
+          ) : (
+            <App {...props} />
+          )}
```

Comment:
In order to render HTML in first paint, we have to remove `PersistGate` at server side. It leads to a well known warning about react dom hydrate error. Mark it here, hope we can find a way to solve it later.